web/src/message_list_data.ts

```diff
@@ -18,6 +18,7 @@ export class MessageListData {
     _local_only: Set<number>;
     _selected_id: number;
     predicate?: (message: Message) => boolean;
+    add_messages_callback?: (messages: Message[]) => void;
```

Comment:
This probably deserves a comment documenting it. It might be that we should move all the documentation comments that currently sit in the constructor to this type declaration at the top; could be a good little prep PR?

Reply:
Yeah, removed.